material_studies/rally/lib/finance.dart

```diff
@@ -0,0 +1,311 @@
+    @required this.amount,
+    @required this.suffix,
+  });
+
+  final Color indicatorColor;
+  final double indicatorFraction;
+  final String title;
+  final String subtitle;
+  final double amount;
+  final Widget suffix;
+
+  @override
+  Widget build(BuildContext context) {
+    return FlatButton(
+      onPressed: () {
+        Navigator.push(
+            context,
+            MaterialPageRoute<FinancialEntityCategoryDetailsPage>(
+                builder: (context) => FinancialEntityCategoryDetailsPage()));
+      },
+      child: SizedBox(
+          height: 68,
+          child: Column(
+            children: <Widget>[
+              Expanded(
+                child: Row(
+                  children: <Widget>[
+                    Padding(
+                      padding: EdgeInsets.only(left: 12, right: 12),
+                      child: VerticalFractionBar(
+                        color: indicatorColor,
+                        fraction: indicatorFraction,
+                      ),
+                    ),
+                    Column(
+                      mainAxisAlignment: MainAxisAlignment.center,
+                      crossAxisAlignment: CrossAxisAlignment.start,
+                      children: <Widget>[
+                        Text(title,
+                            style: Theme.of(context)
+                                .textTheme
+                                .body1
+                                .copyWith(fontSize: 16.0)),
+                        Text(subtitle,
+                            style: Theme.of(context)
+                                .textTheme
+                                .body1
+                                .copyWith(color: RallyColors.gray60))
+                      ],
+                    ),
+                    Spacer(),
+                    Text('\$ ' + Formatters.usd.format(amount),
+                        style: Theme.of(context)
+                            .textTheme
+                            .body2
+                            .copyWith(fontSize: 20.0, color: RallyColors.gray)),
+                    SizedBox(width: 32.0, child: suffix)
+                  ],
+                ),
+              ),
+              Padding(
+                padding: EdgeInsets.only(left: 16, right: 16),
+                child: SizedBox(
+                    height: 1, child: Container(color: Color(0xAA282828))),
+              )
+            ],
+          )),
+    );
+  }
+}
+
+/// Data model for [FinancialEntityCategoryView].
+class FinancialEntityCategoryModel {
+  final Color indicatorColor;
+  final double indicatorFraction;
+  final String title;
+  final String subtitle;
+  final double usdAmount;
+  final Widget suffix;
+
+  const FinancialEntityCategoryModel(
+    this.indicatorColor,
+    this.indicatorFraction,
+    this.title,
+    this.subtitle,
+    this.usdAmount,
+    this.suffix,
+  );
+}
+
+/// Utils for [FinancialEntityCategoryView].
+class FinancialEntityCategoryViews {
```

Comment:
Similar to the last PR, top-level functions would be a more idiomatic way to write these.

Reply:
Done